### ISSUE TITLE
[info] and [warning] messages seem unnecessary

### ISSUE THREAD

**Issue by RianneSchouten** [RianneSchouten/pyampute#6]
If I run a basic instance, there are quite some [info] and [warning] messages that do not seem to be appropriate. 

`n = 1000`
`X = np.random.randn(n, 2)`

`ma = MultivariateAmputation(seed=2022)`
`incomplete_X = ma.fit_transform(X)`

Output:
2022-01-27 10:13:38,113 [INFO] No patterns passed, setting default pattern.
2022-01-27 10:13:38,121 [INFO] No weights passed for some patterns, filling them in per pattern. MCAR: weights are all 0s. MAR: all observed vars have weight 1. MNAR: all missing vars have weight 1.
2022-01-27 10:13:38,121 [WARNING] Indicated weights for incomplete vars for a pattern with MAR. Did you mean MAR+MNAR?
2022-01-27 10:13:38,121 [WARNING] Binary variables (at indices (array([], dtype=int64),)) are indicated to be used in amputation (they are weighted and will be used to calculate the weighted sum score under MAR, MNAR, or MAR+MNAR). This can result in a subset with candidates that all have the same (or almost the same) weighted sum scores.
2022-01-27 10:13:38,122 [WARNING] Categorical variables (at indices (array([1], dtype=int64),)) are indicated to be used in amputation (they are weighted and will be used to calculate the weighted sum score under MAR, MNAR, or MAR+MNAR).These will be forced to be numeric upon calculating sum scores.
2022-01-27 10:13:38,123 [INFO] Enforcing data to be numeric since calculation of weights requires numeric data.
2022-01-27 10:13:38,123 [WARNING] Numpy does not support heterogeneous ndarrays. While enforcing variables involved in imputation to numeric, we have to enforce the whole ndarray passed in.
2022-01-27 10:13:38,129 [INFO] Rounding proportion of missingness to 2 decimal places in order to use lookup table for one of the prespecified score to probability functions.

**I think the following messages should not have occurred:**
2022-01-27 10:13:38,121 [WARNING] Indicated weights for incomplete vars for a pattern with MAR. Did you mean MAR+MNAR?
2022-01-27 10:13:38,121 [WARNING] Binary variables (at indices (array([], dtype=int64),)) are indicated to be used in amputation (they are weighted and will be used to calculate the weighted sum score under MAR, MNAR, or MAR+MNAR). This can result in a subset with candidates that all have the same (or almost the same) weighted sum scores.
2022-01-27 10:13:38,122 [WARNING] Categorical variables (at indices (array([1], dtype=int64),)) are indicated to be used in amputation (they are weighted and will be used to calculate the weighted sum score under MAR, MNAR, or MAR+MNAR).These will be forced to be numeric upon calculating sum scores.
2022-01-27 10:13:38,123 [INFO] Enforcing data to be numeric since calculation of weights requires numeric data.
2022-01-27 10:13:38,123 [WARNING] Numpy does not support heterogeneous ndarrays. While enforcing variables involved in imputation to numeric, we have to enforce the whole ndarray passed in.

**Comment by davzaman** [comment]
I can get rid of these but in that case you can remove the task for @prabhant  to detect which variables are categorical, since this is the only place we use that information. Is that ok?

**Comment by davzaman** [comment]
Should be fixed with 0a607ce, will close if you're happy with the above.

**Comment by RianneSchouten** [comment]
Thank you for your work @davzaman. I actually did not mean to say that the warnings are wrong or should be removed, but that it seems 'wrong' that they occur for this particular example. In this example, I give numerical data with default settings, then why do warnings appear about weights, binary variables, categorical variables and enforcing data to be numeric? I would expect these warnings to only occur when the data is actually binary or categorical.

**Comment by davzaman** [comment]
fixed wtih 7494c1d
